zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbAuthorizationState.java

```diff
@@ -126,37 +128,71 @@ public void update(final long authorizationKey, final AuthorizationRecord author
   @Override
   public void delete(final long authorizationKey) {
     this.authorizationKey.wrapLong(authorizationKey);
-    final var persistedAuthorization =
-        authorizationByAuthorizationKeyColumnFamily.get(this.authorizationKey);
+    final var authorizationToDelete =
+        authorizationByAuthorizationKeyColumnFamily.get(
+            this.authorizationKey, PersistedAuthorization::new);
 
-    // remove the old permissions
-    persistedAuthorization
+    ownerId.wrapString(authorizationToDelete.getOwnerId());
+    ownerType.wrapString(authorizationToDelete.getOwnerType().name());
+
+    final var overlappingAuthorizationKeys =
+        authorizationKeysByOwnerColumnFamily.get(ownerTypeAndOwnerId);
+
+    // For each authorization key get the authorization and check if it matches the persisted
+    // Authorization ResourceType and ResourceMatcher, collecting the matching authorization
+    // permissions in a set.
+    final var sharedPermissionTypes =
+        overlappingAuthorizationKeys.getAuthorizationKeys().stream()
+            .filter(overlappingKey -> overlappingKey != authorizationToDelete.getAuthorizationKey())
+            .map(
+                key -> {
+                  this.authorizationKey.wrapLong(key);
+                  return authorizationByAuthorizationKeyColumnFamily.get(this.authorizationKey);
```

Comment:
💭 I guess this works as long as the stream is sequential. This getter returns the object defined as a field in this class. Getting it twice would mean the data of the first authorization gets overridden with the data of the second one. We could consider adding `PersistedAuthorization::new` to the call to circumvent this. But as it is now it shouldn't be an issue.